plugins/nodes/src/js/pages/nodes/NodeDetailTab.js

```diff
@@ -57,12 +59,24 @@ class NodeDetailTab extends PureComponent {
     });
   }
 
+  hasMaintenanceData() {
+    const { node } = this.props;
+
+    // Detecting whether DCOS already supports maintenance mode.
+    // We might want to remove this flag at some point in the future.
+    return (
+      typeof node.get("drain_info") !== "undefined" ||
+      typeof node.get("deactivated") !== "undefined"
```

Comment:
Why are we checking the type and not just the value?

Reply:
that's a really good question. will change in #3960 as well!